pkg/config/stepmeta.go

```diff
@@ -278,18 +278,21 @@ func (m *StepData) GetContextDefaults(stepName string) (io.ReadCloser, error) {
 			if len(container.Command) > 0 {
 				p["containerCommand"] = container.Command[0]
 			}
-			p["containerName"] = container.Name
-			p["containerShell"] = container.Shell
-			p["dockerEnvVars"] = EnvVarsAsMap(container.EnvVars)
-			p["dockerImage"] = container.Image
-			p["dockerName"] = container.Name
-			p["dockerPullImage"] = container.ImagePullPolicy != "Never"
-			p["dockerWorkspace"] = container.WorkingDir
-			p["dockerOptions"] = OptionsAsStringSlice(container.Options)
-			//p["dockerVolumeBind"] = volumeMountsAsStringSlice(container.VolumeMounts)
+
+			putStringIfNotEmpty(p, "containerName", container.Name)
+			putStringIfNotEmpty(p, "containerShell", container.Shell)
+			putMapIfNotEmpty(p, "dockerEnvVars", EnvVarsAsMap(container.EnvVars))
+			putStringIfNotEmpty(p, "dockerImage", container.Image)
+			putStringIfNotEmpty(p, "dockerName", container.Name)
+			if container.ImagePullPolicy != "" {
```

Comment:
This line here is the bug fix in the narrower sense, isn't it. All the rest is more the less a general improvement in the same direction, but not strongly required in order to fix the bug. Just asking for clarification.

Reply:
yes, without this the resulting boolean in the next line is always `true` if the step parameter is not explicitly configured, as in that case `ImagePullPolicy` is equal to an empty string:
`"" != "Never"` --> `true`